### ISSUE TITLE
LeafPic crash when opening preview from OpenCamera

### ISSUE THREAD

**Issue by alekksander** [apcro/leafpicrevived#1]
Stack trace:  
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.alienpants.leafpic/com.alienpants.leafpic.activities.SingleMediaActivity}: android.view.InflateException: Binary XML file line #22: Binary XML file line #19: Error inflating class TextView
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2798)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2876)
	at android.app.ActivityThread.-wrap11(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1609)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6606)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
Caused by: android.view.InflateException: Binary XML file line #22: Binary XML file line #19: Error inflating class TextView
Caused by: android.view.InflateException: Binary XML file line #19: Error inflating class TextView
Caused by: java.lang.RuntimeException: A 'Iconics.init(context)' has to happen first. Call from your application. Usually this happens via an 'IconicsDrawable' usage.
	at com.mikepenz.iconics.Iconics.a(Unknown Source:329)
	at com.mikepenz.iconics.Iconics.a(Unknown Source:5)
	at com.mikepenz.iconics.Iconics.a(Unknown Source:2)
	at com.mikepenz.iconics.Iconics.a(Unknown Source:5)
	at com.mikepenz.iconics.Iconics$BuilderView.a(Unknown Source:87)
	at com.mikepenz.iconics.utils.IconicsExtensionsKt.a(Unknown Source:22)
	at com.mikepenz.iconics.utils.IconicsExtensionsKt.a(Unknown Source:6)
	at com.mikepenz.iconics.context.IconicsFactory.b(Unknown Source:41)
	at com.mikepenz.iconics.context.IconicsFactory.a(Unknown Source:30)
	at com.mikepenz.iconics.context.InternalLayoutInflater$WrapperFactory2.onCreateView(Unknown Source:21)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:772)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:866)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
	at android.view.LayoutInflater.parseInclude(LayoutInflater.java:995)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:859)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
	at com.mikepenz.iconics.context.InternalLayoutInflater.inflate(Unknown Source:8)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
	at androidx.appcompat.app.AppCompatDelegateImpl.c(Unknown Source:23)
	at androidx.appcompat.app.AppCompatActivity.setContentView(Unknown Source:4)
	at com.alienpants.leafpic.activities.SingleMediaActivity.onCreate(Unknown Source:6)
	at android.app.Activity.performCreate(Activity.java:7203)
	at android.app.Activity.performCreate(Activity.java:7194)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1215)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2751)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2876)
	at android.app.ActivityThread.-wrap11(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1609)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6606)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)

**Comment by apcro** [comment]
Hi,

Can you confirm which version of the app you are using or if you built it from source, and the steps you are following to cause this crash?

**Comment by alekksander** [comment]
of course. this is latest from google store (not seeing Your fork in f-droid repo) – „v0.8-alpha-3”. Open camera version is 1.47.2. 
to reproduce:
→open open camera from shortcut as usual
→click on miniature to open latest photo (album)
→prompt shows up to choose camera roll or leafpic
→select leafpic
→enjoy crash :)
this is stock android 8.1.0

if i open leafpic normally i cannot see this crash, but recently i managed to crash it some other ways. hopefully will report once stumbled upon again. there are also bugs with not being able to zoom and i think few others i cannot recall. should i report each such failures?
thanks.

**Comment by apcro** [comment]
Thanks, I'm looking in to this.

**Comment by apcro** [comment]
I've come back to this and installed OpenCamera (from source) on my device. I've not been able to recreate the issue. That said, since the version you mention I have done additional work regarding how Ionics is called (the exception references `A 'Iconics.init(context)' has to happen first`), so this may have already been fixed.

LeafPic Revived v0.8.4
OpenCamera 1.47.3

If it happens again, please re-open

**Comment by apcro** [comment]
I've now confirmed this using OpenCamera, so I am re-opening this issue.

**Comment by apcro** [comment]
Build version: v0.8.4 
Build date: 1979-11-30 00:00:00 
Current date: 2020-02-19 16:57:31 
Device: HUAWEI STK-LX1 
 
Stack trace:  
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.io.File.getCanonicalPath()' on a null object reference
	at androidx.core.content.FileProvider$SimplePathStrategy.getUriForFile(FileProvider.java:728)
	at androidx.core.content.FileProvider.getUriForFile(FileProvider.java:418)
	at com.alienpants.leafpicrevived.util.LegacyCompatFileProvider.getUri(LegacyCompatFileProvider.java:25)
	at com.alienpants.leafpicrevived.activities.SingleMediaActivity.onOptionsItemSelected(SingleMediaActivity.java:563)
	at android.app.Activity.onMenuItemSelected(Activity.java:3742)
	at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:384)
	at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:219)
	at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
	at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
	at androidx.appcompat.app.ToolbarActionBar$2.onMenuItemClick(ToolbarActionBar.java:64)
	at androidx.appcompat.widget.Toolbar$1.onMenuItemClick(Toolbar.java:207)
	at androidx.appcompat.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:781)
	at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:840)
	at androidx.appcompat.view.menu.SubMenuBuilder.dispatchMenuItemSelected(SubMenuBuilder.java:90)
	at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:158)
	at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:991)
	at androidx.appcompat.view.menu.MenuPopup.onItemClick(MenuPopup.java:128)
	at android.widget.AdapterView.performItemClick(AdapterView.java:321)
	at android.widget.AbsListView.performItemClick(AbsListView.java:1234)
	at android.widget.AbsListView$PerformClick.run(AbsListView.java:3234)
	at android.widget.AbsListView$3.run(AbsListView.java:4210)
	at android.os.Handler.handleCallback(Handler.java:907)
	at android.os.Handler.dispatchMessage(Handler.java:105)
	at android.os.Looper.loop(Looper.java:216)
	at android.app.ActivityThread.main(ActivityThread.java:7625)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)

**Comment by apcro** [comment]
This crash is unrelated to OpenCamera, I just happened across it when looking at the OpenCamera gallery. Moving to a new issue.
